### DISCUSSION TITLE
Bun v0.0.44

### DISCUSSION THREAD

**Discussion by Jarred-Sumner** [oven-sh/bun#50] [Announcements]
To upgrade:

```bash
bun upgrade
```

## Next.js 12

Bun supports Next.js 12 now. Before, it only worked with 11.1.2. This includes support for React 18 (`react@alpha`). React Server Components is not supported yet, but it will be. I promise :)

<img width="600" alt="CleanShot 2021-11-03 at 03 11 49@2x" src="https://user-images.githubusercontent.com/709451/140042287-59e72802-4684-4023-be3e-10f4834963f8.png">

## Other stuff

- (`bun run`) ZSH completions should auto-install correctly now. This will automatically apply when you run `bun upgrade`
- (`bun dev`) Fixed a mistranspilation when using `require` from app code to a bundled module (e.g. `require("relay-devtools")`)
- (`bun bun`) Fixed an issue with symlinked workspace packages when using `alwaysBundled` that prevented them from being correctly imported if enough levels of indirection were added
- Fixed undefined behavior when invalid UTF-8 codepoints were inside source code text printed in an error message
- (`bun-macro-relay`) Worked around a bug that could cause it to not correctly import `graphql` depending on the environment
- (`bun create`) bumped the hardcoded Next.js version set to `12.0.2`

**Full Changelog**: https://github.com/Jarred-Sumner/bun/compare/bun-v0.0.42...bun-v0.0.44

<hr /><em>This discussion was created from the release <a href='https://github.com/Jarred-Sumner/bun/releases/tag/bun-v0.0.44'>Bun v0.0.44</a>.</em>

**Comment by alvarlagerlof** [comment]
I really need to install this and try now.
